scripts/start.sh

```diff
@@ -0,0 +1,345 @@
+    ${KUBECTL} create configmap bootstrap-rules --from-file config/vaults/bootstrap-rules.json
+  }
+
+  declare -a policies=(tuf crypt gateway)
+  for policy in "${policies[@]}"; do
+    ${KUBECTL} get configmap "${policy}-policy" &>/dev/null || {
+      ${KUBECTL} create configmap --from-file "config/vaults/${policy}-policy.hcl" "${policy}-policy"
+    }
+    ${KUBECTL} get secret "${policy}-tokens" &>/dev/null || {
+      ${KUBECTL} create secret generic "${policy}-tokens"
+    }
+  done
+}
+
+create_databases() {
+  local pod
+  pod=$(wait_for_pods mysql)
+  ${KUBECTL} cp "${CWD}/sql" "${pod}:/tmp"
+  ${KUBECTL} exec "${pod}" -- bash -c "mysql -p${DB_PASS} < /tmp/sql/install_plugins.sql || true" 2>/dev/null
+  ${KUBECTL} exec "${pod}" -- bash -c "mysql -p${DB_PASS} < /tmp/sql/create_databases.sql"
+}
+
+init_vault() {
+  local vault=${1}
+  local api="http://localhost:8200/v1"
+
+  if [[ $(http "${api}/sys/health" | jq '.initialized') = false ]]; then
+    local result
+    result=$(http --check-status PUT "${api}/sys/init" \
+      secret_shares:="${VAULT_SHARES}" secret_threshold:="${VAULT_THRESHOLD}")
+    ${KUBECTL} create secret generic "${vault}-init" \
+      --from-literal="root=$(echo "${result}" | jq --raw-output '.root_token')" \
+      --from-literal="keys=$(echo "${result}" | jq --raw-output '.keys[]')"
+  fi
+}
+
+unseal_vault() {
+  local vault=${1}
+  local pod=${2}
+
+  if ${KUBECTL} get secrets "${vault}-init" &>/dev/null; then
+    ${KUBECTL} get secrets "${vault}-init" -o json |
+      jq -r .data.keys |
+      base64 --decode  |
+      awk 'BEGIN {print "export VAULT_ADDR=http://127.0.0.1:8200"}
+                 {print "vault unseal "$0}
+           END   {print "vault status"}' |
+      ${KUBECTL} exec -i "${pod}" sh
+  fi
+}
+
+start_vaults() {
+  create_configs
+
+  for vault in ${VAULTS:-tuf-vault crypt-vault}; do
+    apply_template "templates/vaults/${vault}.tmpl.yaml"
+
+    local pod
+    pod=$(wait_for_pods "${vault}")
+    ${KUBECTL} port-forward "${pod}" 8200:8200 &
+    local pid=$!
+    trap "kill_pid ${pid}" EXIT
+    sleep 3s
+
+    init_vault "${vault}"
+    unseal_vault "${vault}" "${pod}"
+    kill_pid "${pid}"
+
+    apply_template "templates/jobs/${vault}-bootstrap.tmpl.yaml"
+  done
+}
+
+start_weave() {
+  [[ ${SKIP_WEAVE} == true ]] && return 0;
+  apply_template templates/weave
+}
+
+start_ingress() {
+  [[ ${SKIP_INGRESS} == true ]] && return 0;
+  apply_template templates/ingress
+}
+
+start_infra() {
+  apply_template templates/infra
+  wait_for_pods kafka
+  create_databases
+}
+
+start_services() {
+  apply_template templates/services
+  get_credentials
+}
+
+get_credentials() {
+  ${KUBECTL} get secret "user-keys" &>/dev/null && return 0
+
+  ${KUBECTL} proxy --port 8200 &
+  local pid=$!
+  trap "kill_pid ${pid}" EXIT
+  sleep 3s
+
+  local namespace="x-ats-namespace:default"
+  local api="http://localhost:8200/api/v1/namespaces/default/services"
```

Comment:
Would be really nice if this were configurable. Especially the namespace. I don't deploy stuff to "default" in my k8s.

Reply:
Yes, I'll create another PR with env-vars to configure the namespace and proxy port.